src/observable/FromEventPatternObservable.ts

```diff
@@ -31,26 +29,39 @@ export class FromEventPatternObservable<T, R> extends Observable<T> {
   }
 
   protected _subscribe(subscriber: Subscriber<T>) {
-    const addHandler = this.addHandler;
     const removeHandler = this.removeHandler;
-    const selector = this.selector;
 
-    const handler = selector ? function(e: any) {
-      let result = tryCatch(selector).apply(null, arguments);
-      if (result === errorObject) {
-        subscriber.error(result.e);
-      } else {
+    const handler = !!this.selector ? (...args: Array<any>) => {
+      const [ok, result] = this._callSelector(subscriber, args);
```

Comment:
Same as with the other PR, I'm not sure this approach with the additional Array allocation is benefitting us.